sklearn/manifold/_mds.py

```diff
@@ -272,6 +274,72 @@ def smacof(dissimilarities, metric=True, n_components=2, init=None, n_init=8,
         return best_pos, best_stress
 
 
+def svd_scaler(dissimilarities, n_components=2):
+    """
+    Computes multidimensional scaling using SVD algorithm
+
+    Parameters
+    ----------
+    dissimilarities : ndarray, shape (n_samples, n_samples)
+        Pairwise dissimilarities between the points. Must be euclidean.
+    n_components : int, optional, default: 2
+        Number of dimension in which to immerse the dissimilarities.
+
+    Returns
+    ----------
+    embedding : ndarray, shape (n_samples, n_components)
+        Coordinates of the points in a ``n_components``-space.
+
+    stress : float
+        The final value of the stress (sum of squared distance of the
+        disparities and the distances for all constrained points).
+
+    References
+    ----------
+    "An Introduction to MDS" Florian Wickelmaier
+    Sound Quality Research Unit, Aalborg University, Denmark (2003)
+
+    "Metric and Euclidean properties of dissimilarity coefficients"
+    J. C. Gower and P. Legendre, Journal of Classification 3, 5–48 (1986)
+
+    """
+
+    dissimilarities = check_symmetric(dissimilarities, raise_exception=True)
+
+    n_samples = dissimilarities.shape[0]
+
+    # Centering matrix
+    J = np.eye(*dissimilarities.shape) - (1. / n_samples) * (
+        np.ones(dissimilarities.shape))
+
+    # Double centered matrix
+    B = -0.5 * np.dot(J, np.dot(dissimilarities ** 2, J))
+
+    w, V = linalg.eigh(B, check_finite=False)
+
+    # ``dissimilarities`` is Euclidean iff ``B`` is positive semi-definite.
+    # See "Metric and Euclidean properties of dissimilarity coefficients"
+    # for details
+    try:
+        w = _check_psd_eigenvalues(w)
+    except ValueError:
+        raise ValueError("Dissimilarity matrix must be euclidean. "
```

Comment:
See above: I would directly raise the original `ValueError` stating that the matrix is not PSD, or say that "dissimilarities must be computed using a true distance metric. Make sure to pass a Positive Semi-definite matrix, or use "dissimilarity='euclidean' to use the euclidean distance"